File: src/Homeworks/hw6/Calculator.fs
module hw6.Calculator

let tryCalculate (val1: decimal) operation (val2:decimal) =
    match operation with
    |plus -> Ok (val1 + val2)
    |minus -> Ok (val1 - val2)
    |multiply-> Ok (val1 * val2)
    |divide -> match val2 with
                    | 0m -> Error "Division by zero is not possible"
                    | _ -> Ok (val1 / val2)

File: src/Homeworks/hw8/hw8/Calculator/Calculator.cs
using System.Globalization;
using hw8.Interface;

namespace hw8.Calculator
{
    public class Calculator : ICalculator
    {
        public string Plus(double val1, double val2)
            => (val1 + val2).ToString(CultureInfo.InvariantCulture);

        public string Minus(double val1, double val2)
            => (val1 - val2).ToString(CultureInfo.InvariantCulture);

        public string Multiply(double val1, double val2)
            => (val1 * val2).ToString(CultureInfo.InvariantCulture);

        public string Divide(double val1, double val2) =>
            val2 == 0
                ? "divide by zero"
                : (val1 / val2).ToString(CultureInfo.InvariantCulture);
    }
}